kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -207,11 +209,10 @@ public ConfigurationServiceImpl() {
     //
     // ----------------------------------------------------------------
 
-    protected void activate(ComponentContext componentContext) throws InvalidSyntaxException {
+    protected void activate() throws InvalidSyntaxException {
         s_logger.info("activate...");
 
-        // save the bundle context
-        this.m_ctx = componentContext;
+        this.bundleContext = FrameworkUtil.getBundle(ConfigurationServiceImpl.class).getBundleContext();
```

Comment:
I believe the use of `FrameworkUtil` to get `BundleContext` in service components is not as efficient as the normal DI way. This would make this class hard to test in isolation. My preference would be to use the Dependency Inversion way in OSGi.